pip/req/req_set.py

```diff
@@ -439,6 +461,11 @@ def _prepare_file(self, finder, req_to_install):
             # # vcs update or unpack archive # #
             # ################################ #
             if req_to_install.editable:
+                if require_hashes:
+                    raise InstallationError(
+                        'The editable requirement %s cannot be installed when '
+                        'requiring hashes, because there is no single file to '
+                        'hash.' % req_to_install)
```

Comment:
`git fsck` is good except that sometimes genuine commits are bad. [requests](/kennethreitz/requests) has a commit that was accepted through a PR and which has (much later) been determined to be invalid by `git fsck`. There's nothing we can do now without rewriting all of the history since that commit if we were to fix it.